rfcs/0051-stop-transaction-processing.md

```diff
@@ -0,0 +1,206 @@
+greater than the global slot of the block, the block validator will reject
+blocks that define a staged ledger diff different than the empty one.
+
+## Test plan and functional requirements
+
+Unit tests will be added to test the behavior of the block producer and the
+block validator. The following requirements should be tested:
+
+* Block producer
+  * The block producer produces blocks when the stop network slot configuration
+    is set to `None`.
+  * The block producer produces blocks when the stop network slot configuration
+    is set to `<slot>` and the current global slot is less than `<slot>`.
+  * The block producer doesn't produce blocks when the stop network slot
+    configuration is set to `<slot>` and the current global slot is greater or
+    equal to `<slot>`.
+  * The block producer processes transactions and fees as usual when the stop
+    transaction slot configuration is set to `None`.
+  * The block producer processes transactions and fees as usual when the stop
+    transaction slot configuration is set to `<slot>` and the current global
+    slot is less than `<slot>`.
+  * The block producer produces empty blocks (blocks with an empty staged ledger
+    diff) when the stop transaction slot configuration is set to `<slot>` and
+    the current global slot is greater or equal to `<slot>`.
+* Block validator
+  * The block validator validates blocks when the stop network slot
+    configuration is set to `None`.
+  * The block validator validates blocks when the stop network slot 
+    configuration is set to `<slot>` and the current global slot is less than
+    `<slot>`.
+  * The block validator always rejects blocks when the stop network slot
+    configuration is set to `<slot>` and the current global slot is greater or
+    equal to `<slot>`.
+  * The block validator validates blocks as usual when the stop transaction slot
+    configuration is set to `None`.
+  * The block validator validates blocks as usual when the stop transaction slot
+    configuration is set to `<slot>` and the global slot of the block is less
+    than `<slot>`.
+  * The block validator rejects blocks that define a staged ledger diff
+    differently than the empty one when the stop transaction slot configuration
+    is set to `<slot>` and the global slot of the block is greater or equal to
+    `<slot>`.
+* Node/client
+  * The node processes transactions from clients as usual when the stop
+    transaction slot configuration is set to `None`.
+  * The node processes transactions from clients as usual when the stop
+    transaction slot configuration is set to `<slot>` and the current global
+    slot is less than `<slot>`.
+  * The node rejects transactions from clients when the stop transaction slot
+    configuration is set to `<slot>` and the current global slot is greater or
+    equal to `<slot>`.
+
+## Drawbacks
+
+Non-patched nodes or nodes with the configuration overridden will still be able
```

Comment:
More than this, non-patched block-producing nodes will constantly attempt to create a de facto hard fork. In general, the major downside is that this is a 'pre-hard-fork hard-fork'.